include/up-cpp/uri/builder/BuildUResource.h

```diff
@@ -140,12 +136,8 @@ namespace uprotocol::uri {
          * @return 
          */
         auto setRpcRequest(const uint32_t id) -> BuildUResource & {
-            if (0 == id) {
-                spdlog::error("UResource id cannot be 0");
-            } else {
-                setName("rpc");
-                setID(id);
-            }
+            setName("rpc");
+            setID(id);
```

Comment:
If the id is 0, you need to call setRpcResponse() so that the name and instance is also populated correctly (incase someone called this API in lieu of setRpcResponse()

Reply:
If I'm understanding this, the ask would be to undo the changes here in `setRpcRequest` since the request doesn't use ID 0. The _response_ uses ID 0 though, so the changes to `setID()` are needed.

@stevenhartley - is that what you are asking for?

Reply:
Wait, so if someone calls setRpcRequest with an ID of 0, then we "assume" they meant setRpcResponse, and call it for them ?  Do I have that right?

Reply:
My interpretation of the ask is for the only change in this file to be the one on line 98. This change here in `setRpcRequest()` is not needed because a _request_ cannot have an ID of 0.